### DISCUSSION TITLE
Usability enhancements

### DISCUSSION THREAD

**Discussion by Psi58** [GrandOrgue/GoOdf#94] [Ideas]
A couple of ideas to further enhance GOODF:

1. Add option to duplicate a panel.
For use cases (1) where a panel is to be further developed, but the designer still wants to retain the original or (2) a derivative is required.
Currently I manage this by exiting GOODF, making the code changes in a text editor, then returning to GOODF. 
I presume it would be relatively simple to code this into GOODF, maybe activated  by right-clicking on a panel name to present a 'Duplicate this panel' dialog.

2. Add more information to GUI Labels in the editor.
When 'Create an ODF label GUI element' is used more than once each entry in the left-hand navigation is identical ("GUI label"), making it impossible to distinguish between them at that level.
Once there is some text added to the element, could the first few characters be appended to the label? It would make comprehension easier if the editor showed "GUI label (text from first item)", "GUI label (text from second item)", eg.

**Comment by larspalo** [comment]
@Psi58 Yes, both ideas could certainly be implemented. The first suggestion is a completely new feature that needs some testing to make sure that the copying of a whole panel object with all its (different) elements behaves correctly. The second would be just a change of current behaviour so that would likely be the first to do.

**Reply by Psi58** [reply]
Great, thanks! Happy to help with testing if required.

**Reply by larspalo** [reply]
@Psi58 The second Gui Label suggestion is already committed and a "test" build can be found at https://github.com/GrandOrgue/GoOdf/actions/runs/8973883921 if you want to try it. As it's nothing complicated that was just a quick change, but there are some other things I want to do/fix before I make a new "official" release.

**Reply by Psi58** [reply]
That's exactly what I envisaged - thanks.
